src/jvm/test/resources/expected_logs/simple_thread_creation_representation_test_jdk8.txt

```diff
@@ -23,7 +23,7 @@ The following interleaving leads to the error:
 
 Detailed trace:
 | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-|                                                                                    Thread 1                                                                                     |                                                     Thread 2                                                      |
+|                                                                                   Main Thread                                                                                   |                                                     Thread 1                                                      |
```

Comment:
Thread names in the title should be consistent with `Thread#<number>` in the trace. Besides, messages like "switch (reason: waiting for Thread 2 to finish)" should also be consistent with this enumeration. 

I would suggest having separate enumeration strategies (starting from 0 or from 1 depending on the mode) to simplify the logic.

Reply:
I added functionality to specify where object numeration starts for specific classes and made sure the displayed number is based on that for: 
- thread headers 
- switch trace points (also start and join tracepoints)
- object numbers like `Thread#N`